packages/engine-render/src/components/extension.ts

```diff
@@ -36,7 +36,7 @@ export class ComponentExtension<T, U, V> {
 
     type!: U;
 
-    zIndex: number = 0;
+    Z_INDEX: number = 0;
```

Comment:
There is a public `zIndex`, so this should be `protected` in case of causing confusion for outside modules.